### ISSUE TITLE
globs above the current directory are silently ignored

### ISSUE THREAD

**Issue by BrianHicks** [facebook/buck2#164]
If you put a relative path above the current directory in a `glob`, those paths will silently be ignored. I would have expected a warning like the one you see if you tell `sh_test` to run a script above the current directory.

For example, `glob(["../lib/**/*.js"])`

**Comment by ndmitchell** [comment]
See https://buck2.build/docs/legacy/concepts/build-rule/#package-boundaries-and-access-to-source-files - basically a glob can't get at anything not defined in the package - so glob doesn't match against the file system but against the list of files associated with this target. Let me write down some docs.

**Comment by ndmitchell** [comment]
Going to add:

```
    /// The `glob()` function specifies a set of files using patterns. As an example:
    ///
    /// ```
    /// glob(["foo/**/*.h"])
    /// ```
    ///
    /// This call will match all header files in the `foo` directory, recursively.
    ///
    /// You can also pass a named `exclude` parameter to remove files matching a pattern:
    ///
    /// ```
    /// glob(["foo/**/*.h"], exclude = ["**/config.h"])
    /// ```
    ///
    /// This call will remove all `config.h` files from the initial match.
    ///
    /// The `glob()` call is evaluated against the list of files owned by this `BUCK` file.
    /// A file is owned by whichever `BUCK` file is closest above it - so given `foo/BUCK` and
    /// `foo/bar/BUCK` the file `foo/file.txt` would be owned by `foo/BUCK` (and available from
    /// its `glob` results) but the file `foo/bar/file.txt` would be owned by `foo/bar/BUCk`
    /// and _not_ appear in the glob result of `foo/BUCK`, even if you write `glob(["bar/file.txt"])`.
    /// As a consequence of this rule, `glob(["../foo.txt"])` will always return an empty list of files.
    ///
    /// Currently `glob` is evaluated case-insensitively on all file systems, but we expect
    /// that to change to case sensitive in the near future.
```

**Comment by stepancheg** [comment]
> those paths will silently be ignored

That is bad.

**Comment by stepancheg** [comment]
Assigning to myself to add some diagnostics.

**Comment by BrianHicks** [comment]
Yes, thank you for picking up what I was saying. Reading it back, I don't think I was clear enough that the problem is the silence. The boundaries are fine!

**Comment by stepancheg** [comment]
Should be fixed after a5c0b18a5c5de7d74098855e04189da1de288144 and b706cf3264b7d14b2f066921d6f0153e4011c6b5.
